src/pages/resultsView/enrichments/GeneBarPlot.tsx

```diff
@@ -154,37 +154,41 @@ export default class GeneBarPlot extends React.Component<IGeneBarPlotProps, {}>
-                                selectedValue={this.geneQuery}
-                                onSelectedGenesChange={(value, genes, label) => {
-                                    this._geneQuery = value;
-                                    this.selectedGenes = genes;
-                                    this._label = label;
-                                    this.isGeneSelectionPopupVisible = false;
-                                }}
-                                defaultNumberOfGenes={DEFAULT_GENES_COUNT}
-                                containerType={this.props.containerType} />
-                        }
-                        placement="bottomLeft"
-                    >
-                        <div>
-                            <button className="btn btn-default btn-xs">
-                                Select genes
-                            </button>
-                        </div>
-                    </DefaultTooltip>
+            <div>
+                <div style={{ zIndex: 10, position: "absolute", top: "10px", left: "10px" }}>
+                    <div className={styles.ChartControls}>
+                        <strong>{ this._label || this.defaultOption.label}</strong>
+                        <DefaultTooltip
+                            trigger={['click']}
+                            destroyTooltipOnHide={false}
+                            visible={this.isGeneSelectionPopupVisible}
+                            onVisibleChange={(visible) => {
+                                this.isGeneSelectionPopupVisible = visible;
+                            }}
+                            overlay={
+                                <GenesSelection
+                                    options={this.geneListOptions}
+                                    selectedValue={this.geneQuery}
+                                    onSelectedGenesChange={(value, genes, label) => {
+                                        this._geneQuery = value;
+                                        this.selectedGenes = genes;
+                                        this._label = label;
+                                        this.isGeneSelectionPopupVisible = false;
+                                    }}
+                                    defaultNumberOfGenes={DEFAULT_GENES_COUNT}
+                                    containerType={this.props.containerType} />
+                            }
+                            placement="bottomLeft"
+                        >
+                            <div>
+                                <button className="btn btn-default btn-xs">
+                                    Select genes
+                                </button>
+                            </div>
+                        </DefaultTooltip>
+                    </div>
+                </div>
+                <div style={{ zIndex: 10, position: "absolute", top: "10px", right: "10px" }}>
```

Comment:
@alisman Only changed L158 and L191 to align them to left and right respectively.